### DISCUSSION TITLE
Exit 1 not making the server to boot from local disk with UEFI

### DISCUSSION THREAD

**Discussion by kaleal** [ipxe/ipxe#789] [Q&A]
Hello there,

I am trying to boot from local disk returning `exit 1` as documented but it doesn't work.
The server is configured to UEFI.

Boot order is set to:
1. UEFI Network
2. UEFI Hard disk

Setting up the server as UEFI Hard Disk as first option it works just fine.

```
chain --replace https://${dhcp-server}/ipxe?mac=${mac}&arch=${arch}
```

This returns the code below when the server is set to boot from disk.
```
#!ipxe

exit 1
```

But instead of moving on the next boot option defined on UEFI setup it throws an error message.
```
Could not boot: Error 0x00000001 (https://ipxe.org/0000001)
```

Any suggestion?

**Comment by stappersg** [comment]
> Any suggestion?

https://lists.ipxe.org/pipermail/ipxe-devel/2019-April/006644.html

And please change this _general_ in discussion in _Q & A_ discussion, so it can be marked as answered.

**Reply by stappersg** [reply]
Related merge request is https://github.com/ipxe/ipxe/pull/612#issuecomment-1203555426

**Comment by NiKiZe** [comment]
Upgrade your bios/firmware to handle exit in a good way, it it still isn't working then complain to that vendor ;)

**Reply by kaleal** [reply]
Thanks, I will do that.

**Reply by stappersg** [reply]
> >  _other firmware_

> Thanks, I will do that.

When that is not an option, try the `exit 1` on a computer from a different (firmware) vendor.

And please report back.

**Comment by kaleal** [comment]
I upgraded the server firmware today and it didn't resolve it.

Running the command below (from https://github.com/ipxe/ipxe/issues/438) and recompiling ipxe is a real expectation to solve that?
```git revert --strategy resolve 1295b4acff1f2014261c40d9f9d2107ffd668d92```

**Reply by NiKiZe** [reply]
Not sure why you think it would ;)

With exit 1 you tell iPXE to exit and return "error code 1".
If your firmware is "configured" to try the next boot device when it returns from other boot program then this should work. All firmware implementors does not however do this.
There is no specification about how this should work, but we think that any sane firmware implementors should do it that way.
Could you maybe find the creator of your firmware?
If unsure then brand and model of server/mainbord and thru the copyright notices in firmware or flash utilities maybe we can find out.

**Comment by kaleal** [comment]
I managed to make it work with a workaround.

Boot order:
1. BIOS Network
2. UEFI Hard disk

With this the installation is done using BIOS mode and all UEFI related is done in preseed, such as efi partitioning and grub-efi-amd64 installation, then mounting the efi partition and running `grub-install --target=x86_64-efi --no-nvram` on a post-install script.

On the next boot it can handle the `exit` with no issues on BIOS mode, then the system is loaded from hard disk using UEFI. That is a kind of workaround that I wouldn't like to do but it finally worked.

**Reply by NiKiZe** [reply]
Yeah, great you got a workaround, but that's ugly, please, again, get us the requested information about brand etc, maybe we can get them to do something about it, or an explanation.

**Reply by kaleal** [reply]
It is a Supermicro X11DPU running BIOS version 3.8, the very latest available on supermicro website for downloading.
